File: src/main/java/com/openpms/server/ServerExceptionHandler.java
package com.openpms.server;

import com.alibaba.fastjson2.JSONSchemaValidException;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.ControllerAdvice;
import org.springframework.web.bind.annotation.ExceptionHandler;

import java.util.HashMap;
import java.util.Map;

@ControllerAdvice
public class ServerExceptionHandler {
    @ExceptionHandler(JSONSchemaValidException.class)
    public ResponseEntity<?> handleValidationExceptions(JSONSchemaValidException exception) {
        Map<String, String> errors = new HashMap<>();
        errors.put("message", exception.getMessage());
        return ResponseEntity.badRequest().body(errors);
    }
}

File: src/main/kotlin/com/creator/server/ServerExceptionHandler.kt
package com.creator.server

import com.alibaba.fastjson2.JSONSchemaValidException
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.ControllerAdvice
import org.springframework.web.bind.annotation.ExceptionHandler

@ControllerAdvice
class ServerExceptionHandler {
    @ExceptionHandler(JSONSchemaValidException::class)
    fun handleValidationExceptions(exception: JSONSchemaValidException): ResponseEntity<*> {
        val errors: MutableMap<String, String?> = HashMap()
        errors["message"] = exception.message
        return ResponseEntity.badRequest().body<Map<String, String?>>(errors)
    }
}
